src/charm.py

```diff
@@ -96,17 +96,25 @@ def reconcile(self) -> None:
                 "Database configuration missing username, password or URI"
             )
             return
-        logger.info("Matrix Reconciling charm")
-        matrix = self._matrix.reconcile()
+        try:
+            logger.info("Matrix Reconciling charm")
+            matrix = self._matrix.get_matrix()
```

Comment:
I think it would be an advantage to have some unit tests that could test this logic.

Reply:
Yes, agree and these are planned. Note that although tests for this are not part of this PR, the library itself has unit tests (under synapse-operator) and there is an integration test that covers this part to some extent (via an any-charm that is cleverly disguised as a matrix server).

Reply:
For now, I will consider the unit tests in library and the integration test that relates IRC bridge with Matrix, generating a configuration file used by the snap. If this configuration file is not valid, the integration test fails.